hluti1.sql

```diff
@@ -0,0 +1,174 @@
+    house_name TEXT,
+    UNIQUE(kingdom_name, house_name)
+);
+
+-- Define a Common Table Expression (CTE) to simplify the insertion process by handling complex join logic separately
+WITH kingdom_house_mapping AS (
+    SELECT
+        k.gid AS kingdom_id,  -- Assuming 'gid' is a unique identifier for kingdoms
+        h.id AS house_id      -- Assuming 'id' is a unique identifier for houses
+    FROM
+        atlas.kingdoms k
+    FULL OUTER JOIN
+        got.houses h ON k.name = h.region
+)
+-- Insert data into the mapping table, avoiding duplicates for house IDs
+INSERT INTO baratheon.tables_mapping (kingdom_id, house_id)
+SELECT kingdom_id, house_id FROM kingdom_house_mapping
+ON CONFLICT (house_id) DO NOTHING;  -- PostgreSQL-specific UPSERT handling
+
+-- Query to fetch and display all current mappings from the tables_mapping table
+SELECT * FROM baratheon.tables_mapping;
+
+
+
+
+
+
+--liður 2
+
+WITH gagntaek_vorpun AS (
+    -- Mapping locations to houses based on the location's name possibly matching the house's seat or another attribute
+    SELECT
+        l.gid AS location_id,
+        l.name AS location_name,
+        h.id AS house_id,
+        h.name AS house_name,
+        l.summary AS summary
+
+    FROM
+        atlas.locations l
+    JOIN
+        got.houses h ON l.name = ANY(h.seats)  -- Assuming 'seats' could match locations by name
+    WHERE
+        h.region = 'The North'
+),
+X AS (
+select *, regexp_match(summary, house_name) from gagntaek_vorpun
+),
+-- UPSERT into the baratheon.tables_mapping table
+y as (SELECT
+    case when house_name like concat('%',location_name) then 1 else 0 end as better_match,
+*
+      FROM x),
+
+z as (select row_number() over (partition by location_id order by better_match desc) as rank, * from y)
+--INSERT INTO baratheon.tables_mapping (house_id, location_id)
+, g as (select house_id, location_id from z where rank = 1),
+
+-- skoða house_id = 120 af hverju location_id 190 og 301 koma upp
+   -- Ranking based on the direct string length comparison
+
+u AS (
+    SELECT
+        *,
+        -- Calculate the length of the substring match and use it as the matching score
+        CASE WHEN house_name LIKE CONCAT('%', location_name, '%')
+             THEN LENGTH(location_name)
+             ELSE 0
+        END AS match_length
+    FROM x
+),
+w AS (
+    SELECT
+        *,
+        -- Rank locations for each house based on the match length
+        ROW_NUMBER() OVER (PARTITION BY house_id ORDER BY match_length DESC) AS rank
+    FROM u
+)
+
+
+-- Upsert the best match into the mapping table
+INSERT INTO baratheon.tables_mapping (house_id, location_id)
+SELECT
+    house_id,
+    location_id
+FROM w
+WHERE rank = 1
+ON CONFLICT (house_id) DO
+    UPDATE SET location_id = excluded.location_id;
+
+
+
+-- Now select results for 'The North' region
+SELECT
+    h.name AS house_name,
+    h.id AS house_id,
+    l.name AS location_name,
+    l.gid AS location_id
+FROM
+    gagntaek_vorpun otm
```

Comment:
Ég er búin að finna út úr þessu, ef línan væri 

```suggestion 
    baratheon.tables_mapping otm
```

þá keyrir þetta rétt. En þetta hefði átt að sjást í PR